### DISCUSSION TITLE
Make snapshot of page state in Next.js

### DISCUSSION THREAD

**Discussion by AlexanderVishnevsky** [vercel/next.js#14765] [Help] [answered]
Hi everyone!
I faced with some problem while developing Next.js application. It is rather an architectural issue.

I want to switch between routes, but keeping all the states on the page so that I can return to page without loosing it's state. I understand that I need to use initalProps at the top level. But this is only suitable for simple cases. Let's take an example where there're hundreds of states on a page with different levels of hierarchy. Is it possible to make a snapshot of all page states?

I look towards [memo ](https://reactjs.org/docs/react-api.html#reactmemo) from React. Also, I think Redux would help me, but I don't use it in the application at all and it's one more dependency. Perhaps this can be solved using the[ Context Api](https://reactjs.org/docs/context.html).

**Comment by IRediTOTO** [comment]
You mean something like check user login? I use localstorage and api to check user auths. Sorry if my cmt not relate.

**Reply by AlexanderVishnevsky** [reply]
> You mean something like check user login? I use localstorage and api to check user auths. Sorry if my cmt not relate.

No, it's not coonected with auth. It's about different pages like index, about... with saving last state

**Reply by IRediTOTO** [reply]
So use context api is the best way

**Comment by janus-reith** [comment, marked as answer]
@AlexanderVishnevsky you could indeed achieve what you describe by using context, and also share the same context across pages by using a Context Provider in your `_app.js` and `useContext` anywhere you want to access its data.
